plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -117,7 +122,12 @@ export async function createRouter(
         },
       };
 
-      const template = await entityClient.findTemplate(templateName);
+      // Forward authorization from client
+      const user = req.user as BackstageIdentity;
```

Comment:
Are we not going to use a header on a standard form `Authorization: Bearer <idtoken>`? That would support all types of request and payload format, and not pollute the data layer. I see the Rollbar one above did just that

Reply:
Are you referring to adding the token as a parameter here? The difference from the Rollbar plugin is that this code is executed in the backend where the IdentityApi, allowing access to the users token, is not available. The suggestion here is to instead grab the token from the incoming request and forward it along (and forward it as a `Authorization: Bearer...` header in the client http request)

Reply:
I meant in the incoming request from the frontend. You chose to put the token in the payload, I'm wondering if it should be in a header.

Reply:
Ah, now I see what you mean, and the relevance of the question! This line makes an assumption that might not be true, that the backend implementation follows a common pattern that an authentication middleware verifies incoming credentials (e.g. from the Authorization header) and if successful looks up and inserts a user object into the request context. This pattern neatly separates the need for verification and user lookup from plugins use of the user identity to perform actions. That middleware code now resides in [contrib](https://github.com/backstage/backstage/blob/master/contrib/docs/tutorials/authenticate-api-requests.md) though, so this line only works for adopters following this undocumented pattern/requirement.

One option could be to instead use `const token = IdentityClient.getBearerToken(req.headers.authorization);` (wrapped in try/catch as the token might be missing). This will work in the current PR context. @Rugvip had some wishes to keep options open for cookie authentication, but this might be a later refactoring?

Reply:
So maybe `IdentityToken.fromRequest(req)` then? (Just a helper that returns an optional string).

And authorization (a middleware that checks if you are permitted to do a thing) is orthogonal to what I'm looking for here. It may or may not be present, and may or may not read the same token out of the request.

I'm just talking about a mechanism to pass through the incoming token to an outgoing request.